### PR TITLE
Custom user settings

### PR DESCRIPTION
**Pull request by milosevicj**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                <span class="glyphicon glyphicon-home"></span> Back to Dashboard
+              </button>
+              
+            </div>
+          </div>
+          <div class="col-xs-6 text-right">   
+	        <div class="next">
+	          <button class="btn btn-default" type="submit">
+                <span class="glyphicon glyphicon-floppy-disk"></span> Update Settings
+              </button>
```
Comment (stanislavgrujic) [inline src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp]: Please fix formatting.

File: src/main/webapp/resources/core/js/pomodoro-timer.js [hunk 2]
```diff
@@ -10,39 +10,53 @@ var audio = new Audio('/resources/core/sounds/beep.mp3');
...
 $('#longButton').on('click', function() {
-  gHours = 0;
-  gMinutes = 15;
-  gSeconds = 0;
+  longBrakeDuration = $('#longButton').attr('data-value');
+  time = bulkMinutesToTime(longBrakeDuration);
+  
+  gHours = time.hours;
+  gMinutes = time.minutes;
+  gSeconds = time.seconds;
```
Comment (stanislavgrujic) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: Apart from the different css selector, everything else is duplicated. Could you please extract it to a function, and reuse?

File: src/main/java/eu/execom/labs/teamodoro/service/UserService.java [hunk 3]
```diff
@@ -36,13 +53,26 @@ public User findByEmail(String email) {
...
     }
 
+    @Transactional
     public User save(User user) {
         user.setEnabled(true);
         BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
         user.setPassword(encoder.encode(user.getPassword()));
+
+        UserSettings settings = getDefaultUserSettings();
+        user.setSettings(settings);
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/labs/teamodoro/service/UserService.java]: So, whenever you save a user, his settings will be reset to default?

File: src/main/java/eu/execom/labs/teamodoro/service/UserService.java [hunk 4]
```diff
@@ -5,20 +5,34 @@
...
 public class UserService {
 
+    @Value("${app.default.pomodoroDuration}")
+    private long DEFAULT_POMODORO_DURATION;
+
+    @Value("${app.default.shortBreakDuration}")
+    private long DEFAULT_SHORT_BREAK_DURATION;
+
+    @Value("${app.default.longBreatDuration}")
+    private long DEFAULT_LONG_BREAK_DURATION;
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/labs/teamodoro/service/UserService.java]: Default times for pomodoro are:
- 25 minutes pomodoro
- 5 minutes short break
- 15 minuts long break

No need to make them configurable outside of our application, so please just define it as constants.

File: src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	    </div>
+    
+	  </form:form>
+       
+    </div>
+    
+  </div>
+  <%@ include file="/WEB-INF/views/jsp/layout/Footer.jsp"%>
+</body>
+</html>
```
Comment (stanislavgrujic) [inline src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp]: Please add I18N for the page.

**Comment by stanislavgrujic** [comment]
What's going on with this pull request? Will there be a following commit? Please fix merge conflicts.

File: src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp (hunk 5 above)
Comment (milosevicj) [inline src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp]: It have been added

File: src/main/java/eu/execom/labs/teamodoro/entity/User.java [hunk 6]
```diff
@@ -61,6 +65,10 @@
     @OneToMany(mappedBy = "user")
     private Set<Pomodoro> pomodoros = new LinkedHashSet<>();
 
+    @OneToOne(fetch = FetchType.EAGER, optional = false, cascade = CascadeType.PERSIST, orphanRemoval = true)
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/labs/teamodoro/entity/User.java]: FetchType is EAGER by default, so no need to explicitly define it here.

File: src/main/webapp/resources/core/js/pomodoro-timer.js (hunk 2 above)
Comment (milosevicj) [inline src/main/webapp/resources/core/js/pomodoro-timer.js]: That will be part of my next pull request (Refactoring pomodoro-timer.js)

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -23,3 +23,11 @@ mail.username=testingpomodoro
...
 mail.invite.subject=You have been invited to join Pomodoro
 mail.invite.body=You have been invited to team in Pomodoro. Follow the link to proceed: <br> <a href="http://localhost:8080">Pomodoro App</a>
+
+
+
+############### Application Specific Configuration ###################
+
+app.default.pomodoroDuration=25
+app.default.shortBreakDuration=5
+app.default.longBreatDuration=15
```
Comment (stanislavgrujic) [inline src/main/resources/application.properties]: I would say that there properties are not needed anymore.

File: src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+          </div>
+          <div class="col-xs-6 text-right">   
+	        <div class="next">
+	          <button class="btn btn-default" type="submit">
+                <span class="glyphicon glyphicon-floppy-disk"></span>
+                <spring:message code="user.settings.save.button" />
+              </button>
+	        </div>
+	      </div>
+	    </div>
```
Comment (stanislavgrujic) [inline src/main/webapp/WEB-INF/views/jsp/UserSettings.jsp]: Please fix formatting in this div block.

**Pull request merged**